### PR TITLE
DateRangeBox: aria-controls, aria-expanded, aria-owns attributes should be synchronised between start and end inputs

### PR DESCRIPTION
**Pull request by EugeniyKiyashko**

### CODE REVIEW

File: js/ui/date_range_box/ui.multiselect_date_box.js [hunk 1]
```diff
@@ -42,11 +42,36 @@ class MultiselectDateBox extends DateBox {
...
 
     _renderOpenedState() {
-        this._strategy.dateRangeBox.option('opened', this.option('opened'));
+        const { opened } = this.option();
 
-        if(this._isStartDateBox()) {
-            super._renderOpenedState();
+        this._strategy.dateRangeBox.option('opened', opened);
+
+        let context = this;
```
Comment (ksercs) [inline js/ui/date_range_box/ui.multiselect_date_box.js]:
```suggestion
        let startDateBox = this;
```

File: js/ui/date_range_box/ui.multiselect_date_box.js [hunk 2]
```diff
@@ -42,11 +42,36 @@ class MultiselectDateBox extends DateBox {
...
+
+    _setAriaAttributes(startDateBox) {
+        const { opened } = this.option();
+
+        const arias = {
+            'expanded': opened,
+            'controls': startDateBox._getControlsAria(),
+        };
+
+        const ariaOwns = (opened || undefined) && startDateBox._popupContentId;
```
Comment (ksercs) [inline js/ui/date_range_box/ui.multiselect_date_box.js]:
```suggestion
        const ariaOwns = opened && startDateBox._popupContentId;
```

File: js/ui/date_range_box/ui.multiselect_date_box.js (hunk 1 above)
Comment (EugeniyKiyashko) [reply js/ui/date_range_box/ui.multiselect_date_box.js]: https://github.com/DevExpress/DevExtreme/pull/24798/commits/b6ec3c67e7405a0d888545fffd26261ca52ebc77

File: js/ui/date_range_box/ui.multiselect_date_box.js (hunk 2 above)
Comment (EugeniyKiyashko) [reply js/ui/date_range_box/ui.multiselect_date_box.js]: https://github.com/DevExpress/DevExtreme/pull/24798/commits/52462e27c7b444d386e0c21ab873c8fab3d81154

**Review by Zedwag — DISMISSED** [review]

File: testing/tests/DevExpress.ui.widgets.editors/dateRangeBox.tests.js [hunk 3]
```diff
@@ -3893,3 +3894,164 @@ QUnit.module('calendarOptions', moduleConfig, () => {
         });
     });
 });
+
+QUnit.module('Aria accessibility', moduleConfig, () => {
+    QUnit.test('aria-owns attribute on root element', function(assert) {
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets.editors/dateRangeBox.tests.js]: 1. We write about "root element" in tests, but there are also asserts for inputs
2. What we expect in this test? aria-owns should be always undefined?

We can name this test as "aria-owns attribute should be set on root element when popup is opened"
And add an other test "aria-owns on dateBox inputs should always be undefined"

File: testing/tests/DevExpress.ui.widgets.editors/dateRangeBox.tests.js [hunk 4]
```diff
@@ -3893,3 +3894,164 @@ QUnit.module('calendarOptions', moduleConfig, () => {
...
+        this.reinit({
+            opened: true
+        });
+
+        assert.strictEqual($(this.instance.endDateField()).attr('aria-expanded'), 'true', 'aria-expanded on startDateInput');
+        assert.strictEqual($(this.instance.endDateField()).attr('aria-expanded'), 'true', 'aria-expanded on endDateInput');
+
+        this.instance.close();
+
+        assert.strictEqual($(this.instance.startDateField()).attr('aria-expanded'), 'false', 'aria-expanded on startDateInput');
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets.editors/dateRangeBox.tests.js]: - it's already checked in test above
- it does not meet the name

File: testing/tests/DevExpress.ui.widgets.editors/dateRangeBox.tests.js [hunk 5]
```diff
@@ -3893,3 +3894,164 @@ QUnit.module('calendarOptions', moduleConfig, () => {
...
+        });
+    });
+
+    [true, false].forEach(deferRendering => {
+        QUnit.test(`'aria-controls' should be set if deferRendering="${deferRendering}"`, function(assert) {
+            this.reinit({
+                deferRendering,
+            });
+
+            assert.strictEqual($(this.instance.startDateField()).attr('aria-controls'), deferRendering ? undefined : $(this.instance.content()).attr('id'), 'aria-controls on startDateInput');
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets.editors/dateRangeBox.tests.js]: simplify plz

File: testing/tests/DevExpress.ui.widgets.editors/dateRangeBox.tests.js [hunk 6]
```diff
@@ -3893,3 +3894,164 @@ QUnit.module('calendarOptions', moduleConfig, () => {
...
+            const newContouredCellID = getContouredCell().attr('id');
+
+            assert.strictEqual(newContouredCellID === contouredCellID, false, 'countoured cell is changed');
+            assert.strictEqual($(this.instance.startDateField()).attr('aria-activedescendant'), newContouredCellID, 'aria-activedescendant on startDateInput');
+            assert.strictEqual($(this.instance.endDateField()).attr('aria-activedescendant'), newContouredCellID, 'aria-activedescendant on endDateInput');
+
+            this.instance.option('opened', false);
+
+            assert.strictEqual($(this.instance.startDateField()).attr('aria-activedescendant'), newContouredCellID, 'aria-activedescendant on startDateInput');
+            assert.strictEqual($(this.instance.endDateField()).attr('aria-activedescendant'), newContouredCellID, 'aria-activedescendant on endDateInput');
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets.editors/dateRangeBox.tests.js]: Too many asserts. I think we can split this test

Comment (EugeniyKiyashko) [reply testing/tests/DevExpress.ui.widgets.editors/dateRangeBox.tests.js]: https://github.com/DevExpress/DevExtreme/pull/24798/commits/fadda309c6c15b8d9459ef9e5c474605c0ef1a54

File: testing/tests/DevExpress.ui.widgets.editors/dateRangeBox.tests.js (hunk 5 above)
Comment (EugeniyKiyashko) [reply testing/tests/DevExpress.ui.widgets.editors/dateRangeBox.tests.js]: https://github.com/DevExpress/DevExtreme/pull/24798/commits/fadda309c6c15b8d9459ef9e5c474605c0ef1a54

File: testing/tests/DevExpress.ui.widgets.editors/dateRangeBox.tests.js (hunk 4 above)
Comment (EugeniyKiyashko) [reply testing/tests/DevExpress.ui.widgets.editors/dateRangeBox.tests.js]: https://github.com/DevExpress/DevExtreme/pull/24798/commits/fadda309c6c15b8d9459ef9e5c474605c0ef1a54

File: testing/tests/DevExpress.ui.widgets.editors/dateRangeBox.tests.js (hunk 3 above)
Comment (EugeniyKiyashko) [reply testing/tests/DevExpress.ui.widgets.editors/dateRangeBox.tests.js]: https://github.com/DevExpress/DevExtreme/pull/24798/commits/fadda309c6c15b8d9459ef9e5c474605c0ef1a54

**Review by ksercs — DISMISSED** [review]

**Review by ksercs — DISMISSED** [review]

**Review by ksercs — DISMISSED** [review]

**Review by Zedwag — APPROVED** [review]

**Pull request merged**